app/Services/AppointmentService.php

```diff
@@ -13,6 +15,37 @@ public function __construct(AppointmentRepository $repository)
         $this->repository = $repository;
     }
 
+    public function index()
+    {
+        if (request()->user('doctor')) {
+            return $this->repository->filterAppointmentsByStatus();
+        }
+    }
+
+    public function approve()
+    {
+        $appointment = $this->repository->find(request()->appointment);
+        $doctor = request()->user();
+        $this->checkDoctorHasThisAppointment($doctor, $appointment);
+        $this->checkAvailabiltyToApprove($appointment);
+        $this->update(['status' => AppointmentStatus::APPROVED] , $appointment->id);
+        $appointment->patient->notify(new AppointmentNotification($this->repository->find(request()->appointment)));
+    }
+
+    public function checkDoctorHasThisAppointment($doctor, $appointment)
+    {
+        $doctor->appointments->contains($appointment) ? "" : abort(Response::HTTP_METHOD_NOT_ALLOWED);
```

Comment:
`HTTP_METHOD_NOT_ALLOWED` is not used in this context. Please use `BAD REQUEST`

Reply:
done